### ISSUE TITLE
Casting error in asynchronously opening a Realm

### ISSUE THREAD

**Issue by Taco55** [realm/realm-cocoa#4873]
## Goals
Use the newly introduced API to asynchronously open a Realm (which is a really great and useful addition, btw!).

## Expected Results
Correct error handling in case the Realm fails to open

## Actual Results
Error in casting `NSError` to `Realm.Error`:

`Could not cast value of type 'NSError' (0x600000248808) to 'RealmSwift.Realm.Error' (0x104d0dfb8).
`

Which happens at:

```swift
public static func asyncOpen(configuration: Realm.Configuration = .defaultConfiguration,
                                 callbackQueue: DispatchQueue = .main,
                                 callback: @escaping (Realm?, Realm.Error?) -> Void) {
        RLMRealm.asyncOpen(with: configuration.rlmConfiguration, callbackQueue: callbackQueue) { rlmRealm, nsError in
            callback(rlmRealm.flatMap(Realm.init), nsError as! Realm.Error?)
        }
    }
```

with the actual `NSError` being:
`Error Domain=io.realm.sync Code=6 "(null)" UserInfo={message=Operation canceled, underlying=89}`

## Steps to Reproduce
Anything that leads to a `NSError` as above while opening a Realm.

## Version of Realm and Tooling
- Realm Object Server Version: 1.4.1
- Client SDK Version: 2.6.0

**Comment by istx25** [comment]
Hi @Taco55. Glad to hear you're excited about the new asynchronous API. I wanted to let you know that we've received your issue and that someone will review what you've shared and follow-up soon. Thanks! :)

**Comment by jpsim** [comment]
Thanks for reporting this so quickly!

**Comment by jpsim** [comment]
I just released 2.6.1 with a fix.
